### PR TITLE
Fix some other warnings related to double-->float casts

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR fixes some other warnings that appear when WarpX is compiled in single precision.

### CODE REVIEW

File: Source/WarpX.cpp [hunk 1]
```diff
@@ -524,7 +524,7 @@ WarpX::ReadParameters ()
         int quantum_xi_is_specified = queryWithParser(pp, "quantum_xi", quantum_xi_tmp);
         if (quantum_xi_is_specified) {
             double const quantum_xi = quantum_xi_tmp;
-            quantum_xi_c2 = quantum_xi * PhysConst::c * PhysConst::c;
+            quantum_xi_c2 = static_cast<amrex::Real>(quantum_xi * PhysConst::c * PhysConst::c);
```
Comment (ax3l) [inline Source/WarpX.cpp:537]: Just double checking here: does the constant fit into a float for `quantum_xi_c2`?

File: Source/Diagnostics/WarpXIO.cpp [hunk 2]
```diff
@@ -90,7 +90,7 @@ WarpX::InitFromCheckpoint ()
             std::istringstream lis(line);
             int i = 0;
             while (lis >> word) {
-                t_new[i++] = std::stod(word);
+                t_new[i++] = static_cast<Real>(std::stod(word));
```
Comment (ax3l) [inline Source/Diagnostics/WarpXIO.cpp:93]: Just ping-ing @WeiqunZhang to take a look at this file, too - because I am not too familiar with it.

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Merging already, probably safe to go.

File: Source/WarpX.cpp (hunk 1 above)
Comment (lucafedeli88) [reply Source/WarpX.cpp:537]: If I understand correctly, these lines allow a user to re-define the quantity `quantum_xi_c2` in order to enhance Quantum effects. The default value (the one which gives physically correct results) is:
```
    static constexpr auto xi_c2 = static_cast<amrex::Real>( 1.1728865132395492e-35 ); // This should be usable for single precision, though
    // very close to smallest number possible (1.2e-38)
```
So it should be fine to use a float here. Indeed, `quantum_xi_c2` is already a `Real`:
```
Real WarpX::quantum_xi_c2 = PhysConst::xi_c2;
```

File: Source/Diagnostics/WarpXIO.cpp (hunk 2 above)
Comment (lucafedeli88) [reply Source/Diagnostics/WarpXIO.cpp:93]: I've just added @WeiqunZhang to the reviewers list

Comment (ax3l) [reply Source/Diagnostics/WarpXIO.cpp:93]: Seems busy, but I think this is fine :)

**Comment by ax3l** [comment]
@lucafedeli88 sorry for this, there is a little merge conflict that you please need to resolve :)

**Pull request merged**